features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -345,8 +345,14 @@ nsapi_error_t AT_CellularNetwork::open_data_channel()
 {
 #if NSAPI_PPP_AVAILABLE
     tr_info("Open data channel in PPP mode");
+
+#ifdef TARGET_MTS_DRAGONFLY_L471QG
```

Comment:
`AT+CGDATA` is likely supported. Can you check/share AT manual of the cellular module?

Reply:
https://www.u-blox.com/sites/default/files/SARA-R4-SARA-N4_ATCommands_%28UBX-17003787%29.pdf

AT+CGDATA is not supported on the sara r410 with the currently approved firmware. This was confirmed with a ublox fae.

Reply:
I ran the astyle utility on the two file mentioned and above and quite a few lines changed when i diffed. Is this expected.

Reply:
@cedrickkukela-cd Features are not yet astyled. We will do it soon. Please, just use it on new files in this pull request (note, targets folder is being ignored at the moment, but would be nice if they follow as well - targets HAL implementation).

Reply:
@cedrickkukela-cd should this be dynamic? The reason is that maybe TARGET_MTS_DRAGONFLY_L471QG could hold different kind of cellular modules (in the future) and some of those modules might support AT+CGDATA?

For dynamic implementation you could use unsupported flagging, just define  `AT_CellularBase::SupportedFeature` with e.g. flag name `AT_CGDATA` in you CellularDevice and and then in open_data_channel `if (!is_supported(AT_CGDATA))`.

Reply:
Dynamic would be better then there is no need for extra #ifdef's. You can check how it was done with QUECTEL_BG96.cpp and how it was used in AT_CellularInformation::get_serial_number(..

Reply:
@cedrickkukela-cd please see an an example how you could do this at
https://github.com/AriParkkila/mbed-os/commit/c7bd3cbbad5e4286f4c80b2746666535b9fe2e78